### ISSUE TITLE
Version 5.7.1 of BIG-IP PAYG NLB stack does not support AdvancedWAF licensing bundle

### ISSUE THREAD

**Issue by fortunecookiezen** [F5Networks/f5-aws-cloudformation#134]
## Do you already have an issue opened with F5 support?

We have not opened an issue with F5 support, but were directed here by our SE

## Description
There is a bug in  the cloudformation template for **Auto scaling the BIG-IP VE Web Application Firewall in AWS: Existing Stack with PAYG Licensing (Frontend via NLB)** template Version 5.7.1. Template [purports to support](https://github.com/F5Networks/f5-aws-cloudformation/blob/master/aws-bigip-version-matrix.md) Good/Better/Best as well as AdvancedWAF licensing bundles, but the template is hard-coded to only request AMI corresponding to the Best PAYG license bundle. Line 2264 of the template ignores the value of the `F5 Image Name` Parameter

## Template
For bugs, enter the template with which you are experiencing issues below.

https://github.com/F5Networks/f5-aws-cloudformation/blob/master/experimental/autoscale/waf/via-nlb/1nic/existing-stack/payg/f5-payg-autoscale-bigip-waf-nlb.template

## Severity Level
For bugs, enter the bug severity level. Do not set any labels.

Severity: 

**4**

Severity level definitions:
 1. **Severity 1 (Critical)** : Defect is causing systems to be offline and/or nonfunctional. immediate attention is required.
 2. **Severity 2 (High)** : Defect is causing major obstruction of system operations.
 3. **Severity 3 (Medium)** : Defect is causing intermittent errors in system operations.
 4. **Severity 4 (Low)** : Defect is causing infrequent interuptions in system operations.
 5. **Severity 5 (Trival)** : Defect is not causing any interuptions to system operations, but none-the-less is a bug.

**Comment by JeffGiroux** [comment]
You can supply an alternative AMI ID in the 'customImageId' field. See article on how-to here:

https://devcentral.f5.com/s/articles/Using-Cloud-Templates-to-Change-BIG-IP-Versions-AWS

**Comment by fortunecookiezen** [comment]
Thanks for the link! We wound up figuring that out to get over the initial hurdle and the issue with the template. Ongoing, we prefer the vendor maintain that image mapping and we manage the infrastructure from the template, as AdvanceWAF 200Mbps ami id will lifecycle. Otherwise the option should be removed from the template and the documentation clarified to indicate that the template only applies to the Best license bundle and not the others listed here: https://github.com/F5Networks/f5-aws-cloudformation/blob/master/aws-bigip-version-matrix.md, which is misleading to customers and consumes time and money to troubleshoot.

**Comment by JeffGiroux** [comment]
@fortunecookiezen At least you were able to workaround! The intent of the customImageId field serves a couple purposes like feeding your own image or pointing to others. In this case, it had a 3rd purpose...workaround for template code.

Yes, template does have a code Best. PD should fix. Re-pasting your PR example code here so it's consolidated in this ticket.

Regarding - https://github.com/F5Networks/f5-aws-cloudformation/blob/6d342f5e3fae040cfef7304ad759153ad8763104/experimental/autoscale/waf/via-nlb/1nic/existing-stack/payg/f5-payg-autoscale-bigip-waf-nlb.template#L2264

Proposed fix
```
          {
           "Ref": "throughput"
          },
          "Best"
          {
            "Ref": "imageName"
          }
         ]
        }
       ]
```

**Comment by shyawnkarim** [comment]
Thank you for reporting this bug. We are now tracking it internally with ID ESECLDTPLT-2276.

**Comment by shyawnkarim** [comment]
Closing due to age. These legacy templates are now in maintenance mode and are being replaced by our next-generation templates available in the [Cloud Templates 2.0 GitHub repo](https://github.com/F5Networks/f5-aws-cloudformation-v2).
